pkg-manager/core/src/install/index.ts

```diff
@@ -320,6 +321,14 @@ export async function mutateModules (
       )
     }
     const packageExtensionsChecksum = isEmpty(opts.packageExtensions ?? {}) ? undefined : createObjectChecksum(opts.packageExtensions!)
+    const pnpmfilePath = path.join(opts.lockfileDir, '.pnpmfile.cjs')
```

Comment:
The location is configurable: https://pnpm.io/pnpmfile#pnpmfile

The hash should probably be sent to `@pnpm/core` and calculated in the same package that runs `pnpmfile`.